test/extended/util/prometheus/helpers.go

```diff
@@ -134,22 +135,43 @@ func StripLabels(m model.Metric, names ...string) model.LabelSet {
 }
 
 func RunQuery(query, ns, execPodName, baseURL, bearerToken string) (*PrometheusResponse, error) {
-	url := fmt.Sprintf("%s/api/v1/query?%s", baseURL, (url.Values{"query": []string{query}}).Encode())
-	contents, err := GetBearerTokenURLViaPod(ns, execPodName, url, bearerToken)
+	queryUrl := fmt.Sprintf("%s/api/v1/query?%s", baseURL, (url.Values{"query": []string{query}}).Encode())
+	result, err := runQuery(queryUrl, ns, execPodName, bearerToken)
 	if err != nil {
-		return nil, fmt.Errorf("unable to execute query %s: %v", query, err)
+		return nil, fmt.Errorf("error executing query %s: %v", query, err)
+	}
+
+	return result, nil
+}
+
+func RunQueryAtTime(query, ns, execPodName, baseURL, bearerToken string, evaluationTime model.Time) (*PrometheusResponse, error) {
+	queryParams := url.Values{"query": []string{query}, "time": []string{evaluationTime.String()}}
+	queryUrl := fmt.Sprintf("%s/api/v1/query?%s", baseURL, queryParams.Encode())
+	result, err := runQuery(queryUrl, ns, execPodName, bearerToken)
+	if err != nil {
+		return nil, fmt.Errorf("error executing query %s: %v", query, err)
```

Comment:
We tend to avoid starting error messages with "error" because we already know an error occurred as we are wrapping it. I would specify only `executing query`, but since the codebase uses `unable to ...`, it might be a better idea to stick to it.

https://thanos.io/tip/contributing/coding-style-guide.md/#wrap-errors-for-more-context-dont-repeat-failed--there

Reply:
Thanks for the link, updated.